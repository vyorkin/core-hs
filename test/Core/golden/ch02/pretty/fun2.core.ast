[(Name "twice",[Name "f",Name "x"],EAp (EVar (Name "f")) (EAp (EVar (Name "f")) (EVar (Name "x")))),(Name "on",[Name "g",Name "f"],ELam [Name "x",Name "y"] (EAp (EAp (EVar (Name "g")) (EAp (EVar (Name "f")) (EVar (Name "x")))) (EAp (EVar (Name "f")) (EVar (Name "y"))))),(Name "ap",[Name "f",Name "x"],EAp (EVar (Name "f")) (EAp (EVar (Name "f")) (EVar (Name "x"))))]
