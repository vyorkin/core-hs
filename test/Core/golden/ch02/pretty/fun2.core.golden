twice f x = f (f x)
on g f = \ x y. (g (f x)) (f y)
ap f x = f (f x)